source/fundamentals/type-hints.txt

```diff
@@ -1,340 +1,436 @@
+  
+   When you use the ``~bson.son.SON`` document type, you must also specify types
+   for the key and value. The preceding example uses ``SON[str, int]``.
 
 Typed Collection
 ----------------
 
-You can use ``~typing.TypedDict`` (Python 3.8+) when using a well-defined schema for the data in a
-``~pymongo.collection.Collection``. Note that all `schema validation`_ for inserts and updates is done on the server.
-These methods automatically add an "_id" field.
+If you use a well-defined schema for the the data in a
+``~pymongo.collection.Collection``, you can use the ``~typing.TypedDict`` class
+to declare types for the values of the elements in the collection.
 
-.. code-block:: python
+.. important::
+   
+   The ``TypedDict`` class is available only in Python 3.8 and later.
+   To use ``TypedDict`` in earlier versions of Python, install the ``typing_extensions``
+   package.
+
+In the following example, ``Movie`` is an instance of ``TypedDict``. Each ``Movie`` object
+contains two key-value pairs: ``name``, a string key with a string value, and
+``year``, a string key with an integer value.
 
-  >>> from typing import TypedDict
-  >>> from pymongo import MongoClient
-  >>> from pymongo.collection import Collection
-  >>> class Movie(TypedDict):
-  ...     name: str
-  ...     year: int
-  ...
-  >>> client: MongoClient = MongoClient()
-  >>> collection: Collection[Movie] = client.test.test
-  >>> inserted = collection.insert_one(Movie(name="Jurassic Park", year=1993))
-  >>> result = collection.find_one({"name": "Jurassic Park"})
-  >>> assert result is not None
-  >>> assert result["year"] == 1993
-  >>> # This will raise a type-checking error, despite being present, because it is added by PyMongo.
-  >>> assert result["_id"]  # type:ignore[typeddict-item]
-
-This same typing scheme works for all of the insert methods (the ``~pymongo.collection.Collection.insert_one`` method,
-the ``~pymongo.collection.Collection.insert_many`` method, and the ``~pymongo.collection.Collection.bulk_write`` method).
-For ``bulk_write`` both ``~pymongo.operations.InsertOne`` and ``~pymongo.operations.ReplaceOne`` operators are generic.
+.. code-block:: python
+   :emphasize-lines: 1, 4-6
+
+   >>> from typing import TypedDict
+   >>> from pymongo import MongoClient
+   >>> from pymongo.collection import Collection
+   >>> class Movie(TypedDict):
+   ...     name: str
+   ...     year: int
+   ...
+   >>> client: MongoClient = MongoClient()
+   >>> collection: Collection[Movie] = client.test.test
+   >>> inserted = collection.insert_one(Movie(name="Jurassic Park", year=1993))
+   >>> result = collection.find_one({"name": "Jurassic Park"})
+   >>> assert result is not None
+   >>> assert result["year"] == 1993
+
+This same typing scheme works for all of the insert methods
+(the ``~pymongo.collection.Collection.insert_one()`` method,
+the ``~pymongo.collection.Collection.insert_many()`` method, and the
+``~pymongo.collection.Collection.bulk_write()`` method).
```

Comment:
Suggestion: this would read much easier as a list:
```suggestion
 - ``~pymongo.collection.Collection.insert_one()``
 - ``~pymongo.collection.Collection.insert_many()``
 - ``~pymongo.collection.Collection.bulk_write()``
```